s3/src/test/scala/com/meltwater/fawn/s3/S3Spec.scala

```diff
@@ -0,0 +1,352 @@
+package com.meltwater.fawn.s3
+
+import cats.effect._
+import com.meltwater.fawn.common.{AWSCredentials, AWSRegion}
+import net.andimiller.munit.cats.effect.styles.FlatIOSpec
+import org.http4s.client.Client
+import org.http4s.dsl.io._
+import org.http4s.implicits._
+import org.http4s.{EntityDecoder, Header, Headers, HttpRoutes}
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+class S3Spec extends FlatIOSpec {
+
+  implicit def unsafeLogger[F[_]: Sync] = Slf4jLogger.getLogger[F]
+
+  val credentials = AWSCredentials("accesskey", "secretkey")
+  val region      = AWSRegion.`af-south-1`
+
+  val bucket = "example-testing-bucket"
+  val key    = "key-example.txt"
+  val body   = "testing-body-example"
+
+  val uploadId = "upload-id-example"
+
+  val requestIdExample                 = "request-id-example"
+  val genericResponse: GenericResponse = GenericResponse(requestIdExample, Headers.empty)
+
+  val eTagExample = "dcalknjgfrewlknocadvsnlkjsdfalnk"
+
+  test("Test create bucket") {
+    val client = HttpRoutes
+      .of[IO] { case PUT -> Root =>
+        Ok("").map(
+          _.withHeaders(
```

Comment:
prefer `.transformHeaders(_.put(Header(...)))` so it doesn't overwrite existing headers